src/petals/client/inference_session.py

```diff
@@ -144,13 +139,24 @@ def step(
             if next_servers:
                 request_metadata["next_servers"] = next_servers
 
+        request_metadata["tensor_structure"] = tensor_structure
+
+        server_side_inference_schema, kwargs_schema = self.rpc_info["inference_schema"]
+        compression = server_side_inference_schema[0].compression
+        inference_schema = tuple(BatchTensorDescriptor.from_tensor(arg, compression) for arg in input_tensors)
+
+        # TODO: create more explicit way to check servers schema and client's structure
+        assert len(input_tensors) + (is_dummy(hypo_ids) and is_dummy(prompts)) >= len(
```

Comment:
```suggestion
        assert len(input_tensors) + >= len(
```

Why add dummies separately here? Aren't they a part of `input_tensors` already?